modules/imgproc/src/thresh.cpp

```diff
@@ -1530,14 +1533,17 @@ void cv::adaptiveThreshold( InputArray _src, OutputArray _dst, double maxValue,
         return;
     }
 
+    CALL_HAL(adaptiveThreshold, cv_hal_adaptiveThreshold, src.data, src.step, dst.data, dst.step, src.cols, src.rows,
+             maxValue, method, type, blockSize, delta);
+
     Mat mean;
 
     if( src.data != dst.data )
         mean = dst;
 
     if (method == ADAPTIVE_THRESH_MEAN_C)
         boxFilter( src, mean, src.type(), Size(blockSize, blockSize),
-                   Point(-1,-1), true, BORDER_REPLICATE );
+                   Point(-1,-1), true, BORDER_REPLICATE|BORDER_ISOLATED );
```

Comment:
Shouldn't we add `BORDER_ISOLATED` to the Gaussian threshold too?
As I can see `GaussianBlur` will be run on separate _CV_32F_ _Mat_, but implicit stating of this flag can make this fact more clear to the users reading the code and can shorten execution paths in underlying functions (like [this one](https://github.com/opencv/opencv/blob/a9b5233f67d29449a36f9c202b4e07ba11aa6d57/modules/core/include/opencv2/core/private.hpp#L413-L451)).